strawberry/annotation.py

```diff
@@ -131,6 +129,15 @@ def _get_type_with_args(
 
     def resolve(self) -> Union[StrawberryType, type]:
         """Return resolved (transformed) annotation."""
+        if self.__resolve_cache__ is None:
+            self.__resolve_cache__ = self._resolve()
+
+        return self.__resolve_cache__
+
+    def _resolve(self) -> Union[StrawberryType, type]:
+        # TODO: I wonder if this resolve should be creating types?
```

Comment:
Leaving this here for the future 😊

    `resolve` is called every time we get the field type, which, turns out,
    is called a lot of times, even when creating the dataclass, maybe
    there's some opportunity for improvements here, but we would need
    to call `resolve` anyway, so it's probably not worth changing how this
    works